presto-main/src/main/java/io/prestosql/sql/planner/optimizations/AddExchanges.java

```diff
@@ -456,6 +456,10 @@ public PlanWithProperties visitSort(SortNode node, PreferredProperties preferred
         @Override
         public PlanWithProperties visitLimit(LimitNode node, PreferredProperties preferredProperties)
         {
+            if (node.isWithTies()) {
+                throw new IllegalStateException("Unexpected node: LimitNode with ties");
```

Comment:
Why? Everything this method does should apply to limit with ties if it were ever encountered, no?

Reply:
Yes, but I wanted to be sure that LimitNode with ties never reaches this point. The same in the other places you pointed out. Should I support LimitNode with ties there? Is there a way to say "no LimitNodes with ties behind this point"?

Reply:
I guess it's fine to not support that here for now. I'm more concerned about other optimizers like UnaliasSymbolReferences, etc, which tend to scattered all over the place.

Reply:
Could you please specify, in which optimizers I need to support Limit with ties?

Reply:
UnaliasSymbolReferences and PruneUnreferencedOutputs and LimitPushdown (for now, since we're in the process of replacing it with Rules).

We should look into whether we need to do anything for WindowFilterPushdown.